YoYoProject/Utility/Macros.cs

```diff
@@ -32,12 +32,67 @@ private static string LatestRuntimeDirectory
             "BaseProject"
         );
 
+        // nkrapivin: I may be wrong about these though, feel free to edit this section.
+
+        private static string LatestRunnerPath => Path.Combine(
+            LatestRuntimeDirectory,
+            "windows",
+            "Runner.exe"
+        );
+
+        private static string LatestIgorPath => Path.Combine(
+            LatestRuntimeDirectory,
+            "bin",
+            "Igor.exe"
+        );
+
+        private static string LatestAssetCompilerPath => Path.Combine(
+            LatestRuntimeDirectory,
+            "bin",
+            "GMAssetCompiler.exe"
+        );
+
+        private static string LatestWebserverPath => Path.Combine(
+            LatestRuntimeDirectory,
+            "bin",
+            "GMWebServer.exe"
+        );
+
+        private static string LatestCompatibilityLibPath => Path.Combine(
+            LatestRuntimeDirectory,
+            "lib",
+            "compatibility.zip"
+        );
+
+        private static string LatestHtml5RunnerPath => Path.Combine(
+            LatestRuntimeDirectory,
+            "html5",
+            "scripts.html5.zip"
+        );
+
         public static string Expand(string value)
```

Comment:
This is _very much_ not how I would implement this.
A better implementation would be a private static readonly dictionary where you TryGet the key that you substring out. Then replace. That will be _significantly_ more maintainable.

Generally when you find a large if pattern like this, that's the better approach. Especially since it'll be O(1) instead of O(n).

Reply:
Yup, I'll definitely remove these.